src/search/searchUtils.ts

```diff
@@ -106,55 +114,170 @@ export function shouldIndexFile(file: TFile, inclusions: string[], exclusions: s
+/**
+ * Create the pattern settings value from the categorized patterns.
+ * @param tagPatterns - The tag patterns.
+ * @param extensionPatterns - The extension patterns.
+ * @param folderPatterns - The folder patterns.
+ * @param notePatterns - The note patterns.
+ * @returns The pattern settings value.
+ */
+export function createPatternSettingsValue({
+  tagPatterns,
+  extensionPatterns,
+  folderPatterns,
+  notePatterns,
+}: {
+  tagPatterns: string[];
+  extensionPatterns: string[];
+  folderPatterns: string[];
+  notePatterns: string[];
+}) {
+  const patterns = [...tagPatterns, ...extensionPatterns, ...notePatterns, ...folderPatterns].map(
+    (pattern) => encodeURIComponent(pattern)
+  );
+
+  return patterns.join(",");
+}
+
+/**
+ * Match the file path with the tag patterns.
+ * @param filePath - The file path to match.
+ * @param tagPatterns - The tag patterns to match the file path with.
+ * @returns True if the file path matches the tags, false otherwise.
+ */
+function matchFilePathWithTags(filePath: string, tagPatterns: string[]): boolean {
+  if (!tagPatterns || tagPatterns.length === 0) return false;
+
+  const file = app.vault.getAbstractFileByPath(filePath);
+  if (file instanceof TFile) {
+    const tags = getTagsFromNote(file);
+    if (tagPatterns.some((pattern) => tags.includes(stripHash(pattern)))) {
       return true;
     }
   }
+  return false;
+}
+
+/**
+ * Match the file path with the extension patterns.
+ * @param filePath - The file path to match.
+ * @param extensionPatterns - The extension patterns to match the file path with.
+ * @returns True if the file path matches the extensions, false otherwise.
+ */
+function matchFilePathWithExtensions(filePath: string, extensionPatterns: string[]): boolean {
+  if (!extensionPatterns || extensionPatterns.length === 0) return false;
+
+  // Extract the file extension without the dot
+  const fileExtension = filePath.split(".").pop()?.toLowerCase() || "";
+
+  // Match if the file extension matches any of the patterns
+  return extensionPatterns.some((ext) => ext.slice(2) === fileExtension);
+}
+
+/**
+ * Match the file path with the folder patterns.
+ * @param filePath - The file path to match.
+ * @param folderPatterns - The folder patterns to match the file path with.
+ * @returns True if the file path matches the folders, false otherwise.
+ */
+function matchFilePathWithFolders(filePath: string, folderPatterns: string[]): boolean {
+  if (!folderPatterns || folderPatterns.length === 0) return false;
+
+  // Normalize path separators to forward slashes to ensure cross-platform compatibility
+  const normalizedFilePath = filePath.replace(/\\/g, "/");
+
+  return folderPatterns.some((pattern) => {
+    // Normalize pattern path separators and remove trailing slashes
+    const normalizedPattern = pattern.replace(/\\/g, "/").replace(/\/$/, "");
+
+    // Check if the path starts with the pattern
+    return (
```

Comment:
This is slightly different from the previous InPathFileList function behavior, which it not only checked file path starts with the folder pattern, but also includes and ends with the folder pattern. I make it only keeps the startWith because it aligns with obsidian app folder exclusion behavior